app_dart/lib/src/request_handlers/vacuum_github_commits.dart

```diff
@@ -0,0 +1,112 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:gcloud/db.dart';
+import 'package:github/github.dart';
+import 'package:meta/meta.dart';
+import 'package:truncate/truncate.dart';
+
+import '../datastore/config.dart';
+import '../model/appengine/commit.dart';
+import '../request_handling/api_request_handler.dart';
+import '../request_handling/authentication.dart';
+import '../request_handling/body.dart';
+import '../service/datastore.dart';
+import '../service/github_service.dart';
+import '../service/scheduler.dart';
+
+/// Query GitHub for commits from the past day and ensure they exist in datastore.
+@immutable
+class VacuumGithubCommits extends ApiRequestHandler<Body> {
+  const VacuumGithubCommits(
+    Config config,
+    AuthenticationProvider authenticationProvider, {
+    @required this.scheduler,
+    @visibleForTesting this.datastoreProvider = DatastoreService.defaultProvider,
+  })  : assert(datastoreProvider != null),
+        super(config: config, authenticationProvider: authenticationProvider);
+
+  final DatastoreServiceProvider datastoreProvider;
+
+  final Scheduler scheduler;
+
+  @override
+  Future<Body> get() async {
+    final DatastoreService datastore = datastoreProvider(config.db);
+
+    for (String repository in Config.schedulerSupportedRepos) {
+      final String owner = repository.split('/').first;
+      final String repo = repository.split('/').last;
+      final RepositorySlug slug = RepositorySlug(owner, repo);
+      await _vacuumRepository(slug, datastore: datastore);
+    }
+
+    return Body.empty;
```

Comment:
always returning Body.empty?

Reply:
This is a cron utility function where we only care if it passed or not (HTTP 200 versus 500). The information is still kept in the logs if we need to audit a run.

Reply:
If the return value doesn't matter, can we remove it, to simplify?

Reply:
No as `RequestHandler` asserts the body is not null.